### PR TITLE
Added controller for card;

### PR DESCRIPTION
**Pull request by RomanKlishch**
- added methods to controller;
- added tests

### CODE REVIEW

File: src/main/java/com/greetingcard/web/controller/GetCardsController.java [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
 import javax.servlet.http.HttpSession;
 import java.util.List;
+import java.util.Map;
 
 @Slf4j
 @RestController
-@RequestMapping(value = "/api/v1")
+@RequestMapping(value = "/api/v1", produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
 public class GetCardsController {
     private CardService cardService;
```
Comment (nomarchia) [inline src/main/java/com/greetingcard/web/controller/GetCardsController.java:26]: Для чего здесь отдельній класс, почему нельзя было этот метод вставить в CardController? Он использует тот же самый cardService или свой собственный?

File: src/test/java/com/greetingcard/web/controller/CardControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Test
+    void getCard() throws Exception {
+        User user = User.builder().id(2).build();
+        Card card1 = Card.builder().name("card1").user(user).build();
+        when(cardService.getCardAndCongratulationByCardId(1, 2)).thenReturn(card1);
+        mockMvc.perform(get("/api/v1/card")
+                .param("id", "1")
+                .sessionAttr("user", user)
```
Comment (Yuliia-Sh) [inline src/test/java/com/greetingcard/web/controller/CardControllerTest.java:46]: shouldn't be get("/api/v1/card/1" -?

Comment (RomanKlishch) [reply src/test/java/com/greetingcard/web/controller/CardControllerTest.java:46]: shouldn't be get("/api/v1/card?id=1"

File: src/test/java/com/greetingcard/dao/jdbc/JdbcCardDaoITest.java [hunk 3]
```diff
@@ -158,6 +176,16 @@ void deleteCardById() {
         assertNull(actualCard);
     }
 
+    @Test
+    @DisplayName("Delete card when user does not has access")
```
Comment (AlexKievDeveloper) [inline src/test/java/com/greetingcard/dao/jdbc/JdbcCardDaoITest.java:180]: Deleting

**Review by AlexKievDeveloper — APPROVED** [review]

File: src/main/java/com/greetingcard/web/controller/CardController.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import javax.servlet.http.HttpSession;
+import java.util.Map;
+
+@Slf4j
+@RestController
+@RequestMapping(value = "api/v1/card", produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
+public class CardController {
+    private CardService cardService;
+
+    @Autowired
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/web/controller/CardController.java:27]: Why you set annotation @Autowired over Constructor

File: src/main/java/com/greetingcard/web/controller/CardController.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    public CardController(CardService cardService) {
+        this.cardService = cardService;
+    }
+
+    @GetMapping()
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/web/controller/CardController.java:32]: Why you set "()" ?

File: src/main/java/com/greetingcard/web/controller/CardController.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            return ResponseEntity.status(HttpServletResponse.SC_OK).body(card);
+        }
+    }
+
+    @PostMapping
+    public ResponseEntity<Object> createCard(@RequestBody Card card, HttpSession session) throws JsonProcessingException {
+        log.info("Creating card request");
+        int lenght = card.getName().length();
+        if (lenght<=0 || lenght>250){
+            throw new IllegalArgumentException("Name is short or too long");
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/web/controller/CardController.java:53]: How do we get here value less than zero? 
Maybe we will do two checks: lenght==0 and message You need put name, lenght>250 and message Name is too long. Name must be less than 250 symbols.

File: src/main/java/com/greetingcard/web/servlet/card/CardServlet.java [hunk 7]
```diff
@@ -22,7 +23,6 @@
     @Autowired
     private CardService cardService;
 
-    @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         log.info("Get card request");
         String[] path = request.getPathInfo().split("/");
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/web/servlet/card/CardServlet.java:28]: We must delete servlets, which was remade to controllers

File: src/test/java/com/greetingcard/web/controller/CardControllerTest.java (hunk 2 above)
Comment (OleksandrGubariev) [reply src/test/java/com/greetingcard/web/controller/CardControllerTest.java:46]: I think, it should be as Julia says

Comment (AlexKievDeveloper) [reply src/test/java/com/greetingcard/web/controller/CardControllerTest.java:46]: We expect to get the value ​​of the id parameter from the path(@RequestParam long id).  - "/api/v1/card?id=1"
We are not expecting a path variable value (@PathVariable("id"))                                     - "/api/v1/card/1"

public ResponseEntity<Object> getCard(HttpSession session, @RequestParam long id)  {....
So I think Roma is right.

Comment (RomanKlishch) [reply src/test/java/com/greetingcard/web/controller/CardControllerTest.java:46]: This was not my idea. I took it from our file in google driver

**Review by Yuliia-Sh — APPROVED** [review]

**Review by OleksandrGubariev — APPROVED** [review]

**Pull request merged**